src/bmsg.c

```diff
@@ -603,70 +603,56 @@ int main (int argc, char *argv [])
     }
     else
     if (streq (bmsg_command, "request")) {
-
         char *agent_name = argv[argn];
         if (!agent_name)
-            die ("missing agent name", NULL);
+             die ("missing agent name", NULL);
 
         argn ++;
         char *subject = argv[argn];
         if (!subject)
-            die ("missing subject", NULL);
-
-        mlm_client_set_producer (client, subject);
-
-        zmsg_t *msg = zmsg_new ();
-        zmsg_addstr (msg, "GET");
+             die ("missing subject", NULL);
 
         argn ++;
+        zmsg_t *request = zmsg_new ();
         // Process additional arguments
         while (argv [argn]) {
-            char *add_arg = argv [argn];
-            zmsg_addstr (msg, add_arg);
-            if ( verbose )
-                zsys_debug ("param: '%s'", add_arg);
-            argn ++;
-        }
-        if ( verbose ) {
-            zsys_debug ("Subject: %s", subject);
-            zmsg_print (msg);
+             char *add_arg = argv [argn];
+             zmsg_addstr (request, add_arg);
+             if ( verbose )
+                 zsys_debug ("param: '%s'", add_arg);
+             argn ++;
         }
-        int rv = mlm_client_sendto (client, agent_name, subject, NULL, 5000, &msg);
+        int rv = mlm_client_sendto (client, agent_name, subject, NULL, 5000, &request);
         if (rv != 0)
-            die ("Request failed", NULL);
+             die ("Request failed", NULL);
 
-        zmsg_destroy (&msg);
+        zmsg_destroy (&request);
 
         // Listen for the result
         zpoller_t *poller = zpoller_new (mlm_client_msgpipe (client), NULL);
         zsock_t *which = (zsock_t *) zpoller_wait (poller, timeout);
         if (which == mlm_client_msgpipe (client)) {
-            msg = mlm_client_recv (client);
+            zmsg_t *recv = mlm_client_recv (client);
+            // Check if positive
+            char *okfail = zmsg_popstr (recv);
+            if ( okfail && streq (okfail, "OK"))
```

Comment:
If you want a generic reply solution, you can't expect that an "OK" would belong to the reply.